src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java

```diff
@@ -70,31 +71,38 @@ public boolean canHandle(
             } else {
                 Person person = selectedApplicant.get().getPerson();
                 ApplicationStatus applicationStatus = selectedApplicant.get().getApplicationStatus();
-                asylumCase.write(AsylumCaseDefinition.HOME_OFFICE_SEARCH_STATUS, "SUCCESS");
-                HomeOfficeCaseStatus selectedMainApplicant = selectedApplicant.get();
-                selectedMainApplicant.setDisplayDateOfBirth(
-                    HomeOfficeDateFormatter.getPersonDateOfBirth(
-                        person.getDayOfBirth(), person.getMonthOfBirth(), person.getYearOfBirth())
-                );
-                selectedMainApplicant.setDisplayDecisionDate(
-                    HomeOfficeDateFormatter.getIacDateTime(applicationStatus.getDecisionDate()));
-                selectedMainApplicant.setDisplayDecisionSentDate(
-                    HomeOfficeDateFormatter.getIacDateTime(applicationStatus.getDecisionCommunication().getSentDate())
-                );
-
-                Optional<HomeOfficeMetadata> metadata = selectMetadata(applicationStatus.getHomeOfficeMetadata());
-                if (metadata.isPresent()) {
-                    selectedMainApplicant.setDisplayMetadataValueBoolean(
-                        ("true".equals(metadata.get().getValueBoolean())) ? "Yes" : "No"
+                if (isNull(person) || isNull(applicationStatus)) {
```

Comment:
@divisathyan This is good as a generic message. But could we not check the error section of the response for any error code and corresponding message and derive a meaningful message to user based on that. I think that would be more accurate of the issue. What do you think?

Reply:
Error message change will be part of RIA-3390